### DISCUSSION TITLE
Api Endpoint Wallets (POST)

### DISCUSSION THREAD

**Discussion by skeutmeier** [MrNaif2018/bitcart#353] [General]
Hi,
Im trying to add an XPUB wallet through the api. The correct swagger API Endpoint for that is a http-post to /wallets

Unfortunately, this does not work. The API does not answer on that endpoint, but tries to redirect to /wallets (GET), which in turn correctly returns a list of wallets.

I can POST to other endpoints though. (for example to /invoices)
Is that a known issue? Did somebdy experienced that behaviour before?

Regarding that redirect, I'm also wondering if 
a) Is there a way to see nginx logs?
b) is there a way to update bitcartcc through the commandline? The button in the webinterface doesn't work for me. Thank you

**Comment by MrNaif2018** [comment]
API shouldn't redirect anywhere. How did you deploy? Custom nginx deployment maybe? Does it work from admin panel to add a wallet?

**Reply by skeutmeier** [reply]
Yes,
I can add wallets through the admin panel.


This is my install script:

```
apt-get update && apt-get install -y git
if [ -d "bitcart-docker" ]; then echo "existing bitcart-docker folder found, pulling instead of cloning."; git pull; fi
if [ ! -d "bitcart-docker" ]; then echo "cloning bitcart-docker"; git clone https://github.com/bitcartcc/bitcart-docker bitcart-docker; fi
export BITCART_HOST=example.com
export BITCART_CRYPTOS=btc,bch,ltc
export BITCART_ADDITIONAL_COMPONENTS=tor
cd bitcart-docker
./setup.sh

```

**Reply by skeutmeier** [reply]
OK, nevermind. I found update.sh.

Works perfectly fine.

But the update did not solve the redirect problem

Do you have any hints how I can see access logs?
thank you

**Reply by MrNaif2018** [reply]
Try sending POST request to the URL you get redirected to

**Reply by skeutmeier** [reply]
it is the same.

/wallets

**Reply by skeutmeier** [reply]
I was able to retrieve the access log.
It confirmes what I experience.

`"POST /api/wallets/ HTTP/2.0" 307 0 "-" "-"`

nginx is sending a http code 307  - temporary redirect

**Reply by MrNaif2018** [reply]
And where does it redirect? To /api/wallets? Could you share the domain name with me to check? You can e.g. send me in DMs on telegram

**Reply by skeutmeier** [reply]
thank you for offering. Would have done it, but luckily i found the problem.

thank you

**Comment by skeutmeier** [comment]
ok, I found the problem

I constructed the URL in a dynamic way. The way I did it added a / at the end, so I posted to /wallets/ instead of /wallets

I didn't know that this could case a problem.
Is that a desired behaviour?



Anyways. It is solved (for me) now. Thank you for your patience

**Reply by MrNaif2018** [reply]
That's exactly what I meant by sending request to where it redirects. Yes, it's expected and that's what I was suspecting

**Reply by skeutmeier** [reply]
ah, ok. thank you for clarifying
